portfolio-project-react/src/components/WorkExp/WorkExp.jsx

```diff
@@ -0,0 +1,206 @@
+import { useEffect, useState } from "react";
+
+import {
+  CLOSE,
+  ERROR_JOB_DESC,
+  EXP_HEADING,
+  HEADING_FILL_DETAILS,
+  LABEL_ADD_EXP,
+  PLACEHOLDER_COMP_NAME,
+  PLACEHOLDER_JOB_dESC,
+  SAVE,
+} from "./constants";
+
+import {
+  LABEL_CURR_EMP,
+  LABEL_START_DATE,
+  LABEL_END_DATE,
+} from "../../constants";
+
+import WorkExpItem from "../WorkExpItem/WorkExpItem";
+import styles from "./WorkExp.module.css";
+import moment from "moment";
+
+const WorkExp = ({ isEdit }) => {
+  const [showExperienceForm, setShowExperienceForm] = useState(false);
+  const [expList, setExpList] = useState([]);
+  const [compName, setCompName] = useState("");
+  const [startDate, setStartDate] = useState("");
+  const [endDate, setEndDate] = useState("");
+  const [isCurrentEmp, setIsCurrentEmp] = useState(false);
+  const [jobDescription, setJobDescription] = useState("");
+
+  const SubmitWorkExpForm = (e) => {
+    e.preventDefault();
+    if (!isCurrentEmp && startDate > endDate) {
+      alert("incorrect end date.");
+    } else {
+      const newWorkExp = {
+        compName,
+        startDate,
+        endDate,
+        isCurrentEmp,
+        jobDescription,
+      };
+
+      setExpList([...expList, newWorkExp]);
+    }
+    setShowExperienceForm(false);
+    alert("New Work Experience data is added.");
+  };
+
+  const deleteExperienceItem = (indexPassed) => {
+    const newList = expList.filter((item, index) => index !== indexPassed);
+
+    setExpList(newList);
+  };
+
+  const updateExperienceItem = (
+    itemIndex,
+    name,
+    startDate,
+    endDate,
+    isCurrentEmp,
+    jobDesc
+  ) => {
+    console.log(
+      "Recieved info is :",
+      itemIndex,
+      name,
+      startDate,
+      endDate,
+      isCurrentEmp,
+      jobDesc
+    );
```

Comment:
is this function really needed? its just logging the arguments to the console. is it part of the task?

Reply:
Work experience item update related logic is also implemented in the same function.